### ISSUE TITLE
Suggestion for redesign

### ISSUE THREAD

**Issue by egonelbre** [Deleplace/programming-idioms#48]
Here's a suggestion for a redesign:
![2015-12-31 15_11_02-handle invalid argument](https://cloud.githubusercontent.com/assets/192964/12064640/ee049b14-afd0-11e5-856f-9732efe55a19.png)

First tabbed interfaces are modal and hence more annoying to use, see "The Humane Interface". I.e. the main interface would be nicer to use if there are no tabs. You can use anchors to jump to the appropriate code example.

Of course adjust as needed. HTML/CSS, mobile friendly: [programming-idioms.zip](https://github.com/Deleplace/programming-idioms/files/75545/programming-idioms.zip)

**Comment by egonelbre** [comment]
Also should be easy to adjust for printing. Remove the header and voting add some no-break and scrolling.

**Comment by Deleplace** [comment]
That's amazing, I didn't know you had such talents in addition to CS knowledge and Go skills!
We can definitely try to make something out of this proposal.
I totally agree that the existing design is somewhat clunky, and was even prepared to put money on the table to have someone redesign it.

I really like your clean, uncluttered design approach.
The logo looks good, especially for the short time it took you to create. The hexagon wheel explicitly refers to the moto "don't reinvent the wheel".
The voting controls are those of StackOverflow, which is good for affordance and readability, but not an obligation.

We can easily enough create a new branch, throw the new draft CSS in it, and deploy it to an alternative "version" on AppEngine. I'm on it as soon as I have 2 hours for that!

**Comment by egonelbre** [comment]
Forgot to mention, I probably don't have time to properly develop the full CSS, (and I don't know your skill level):
* http://learnlayout.com/
* https://bradfrost.github.io/this-is-responsive/patterns.html
* http://learn.shayhowe.com/html-css/
* http://learn.shayhowe.com/advanced-html-css/

Trying to use bootstrap with nice looking sites will eventually be more time-consuming that manually writing the code. Of course some smaller css frameworks (e.g. http://getskeleton.com/), sometimes make sense, mainly the grid handling part.

Of course, I can still answer questions and suggest design approaches.

**Comment by egonelbre** [comment]
I happened to be thinking about the voting controls. Remove the downvote -- people tend to down vote things that they don't like, which for cross-language site isn't a good idea. It's probably just to keep "star", which means, it was useful to me.

**Comment by Deleplace** [comment]
Good idea. It turns outs that I can already disable the downvoting with a conf toggle, without redeploying. But I'll wait a little because an dangling [ + ] button is confusing, indeed I'll probably turn it into a star.

**Comment by Deleplace** [comment]
http://redesign.programming-idioms-alt.appspot.com/random-idiom

Still a lot of work to do in order to integrate new look&feel and preserve features, but it's a start.
